command/config/write/config_write.go

```diff
@@ -132,6 +98,124 @@ func (c *cmd) Run(args []string) int {
 	return 0
 }
 
+func parseConfigEntry(data string) (api.ConfigEntry, error) {
+	// parse the data
+	var raw map[string]interface{}
+	if err := hcl.Decode(&raw, data); err != nil {
+		return nil, fmt.Errorf("Failed to decode config entry input: %v", err)
+	}
+
+	return newDecodeConfigEntry(raw)
+}
+
+func newDecodeConfigEntry(raw map[string]interface{}) (api.ConfigEntry, error) {
+	var entry api.ConfigEntry
+
+	kindVal, ok := raw["Kind"]
+	if !ok {
+		kindVal, ok = raw["kind"]
+	}
+	if !ok {
+		return nil, fmt.Errorf("Payload does not contain a kind/Kind key at the top level")
+	}
+
+	if kindStr, ok := kindVal.(string); ok {
+		newEntry, err := api.MakeConfigEntry(kindStr, "")
+		if err != nil {
+			return nil, err
+		}
+		entry = newEntry
+	} else {
+		return nil, fmt.Errorf("Kind value in payload is not a string")
+	}
+
+	var (
+		// skipWhenPatching should contain anything that legitimately contains a
+		// slice of structs when decoded.
+		skipWhenPatching  []string
+		translateKeysDict map[string]string
+	)
+
+	switch entry.GetKind() {
+	case api.ProxyDefaults:
```

Comment:
Does this mean we don't do any snake case translation for the proxy defaults? I may need to add it then.

Reply:
There was no prior art here, so probably no. There were no fields beyond `Config` present so unless you wanted to specially handle `Config` this seemed ok.

Reply:
And I think mapstructure was already going to allow differences in upper vs lowercase